### PR TITLE
add cpu embedding matrix to optimizer

### PR DESCRIPTION
**Pull request by adolphk-yk**
1. add cpu embedding matrix to optimizer
2. change clip_grad_norm process
3. add embedding parameters statistics

### CODE REVIEW

File: LearningMachine.py [hunk 1]
```diff
@@ -259,7 +259,12 @@ def train(self, optimizer, loss_fn):
                 all_costs.append(loss.item())
                 optimizer.zero_grad()
                 loss.backward()
-                torch.nn.utils.clip_grad_norm_(self.model.parameters(), self.conf.clip_grad_norm_max_norm)
+                if self.conf.use_clip_grad_norm:
```
Comment (woailaosang) [inline LearningMachine.py]: I think, to decrease arguments, don't use conf.use_clip_grad_norm. make "conf.clip_grad_norm_max_norm==-1" represent "not use clip_grad_norm".

Comment (woailaosang) [reply LearningMachine.py]: and changing parameters usage requires update the corresponding place in the Tutorail.md and Tutorial_zh_CN.md

File: block_zoo/Embedding.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 from block_zoo.embedding import *
 import copy
 import logging
+import itertools
```
Comment (woailaosang) [inline block_zoo/Embedding.py]: where use it ?

**Comment by ericwtlin** [comment]
self.embeddings[input_cluster].to(torch.device("cuda" if torch.cuda.is_available() else "cpu"))
logging.info("The embeddings[%s]'s weight is on GPU now, you can modify the weight_on_gpu parameter to change embeddings weight device" % input_cluster)

The log would be wrong when torch.cuda.is_available() == False. Please fix it. @adolphk-yk

**Pull request merged**